hal/ia32/cpu.c

```diff
@@ -255,10 +258,25 @@ void hal_longjmp(cpu_context_t *ctx)
 }
 
 
-void hal_jmp(void *f, void *kstack, void *stack, int argc)
+void hal_jmp(void *f, void *kstack, void *ustack, size_t kargc, const arg_t *kargv)
 {
+	/* We support passing at most 4 args on every architecture. */
+	struct stackArg args[4];
+	size_t i;
+
+	if (kargc > 4) {
```

Comment:
Perhaps an assert too, so it will be easier to find out when inevitably someone pass more than 4?

Reply:
I wanted to but iirc lib should not be used in hal